gitter.py

```diff
@@ -297,9 +299,12 @@ def background():
                 else:
                     log.debug('Received keep-alive on %s', room.name)
 
-        t = threading.Thread(target=background)
-        t.daemon = True
-        t.start()
+        if room._uri not in self.joined_rooms:
```

Comment:
You need to make sure `self.joined_rooms` is never accessed simultaneously. For an example, see how `threading.Lock()` is used inside `errbot/backends/irc.py`.